metricflow/model/objects/data_source.py

```diff
@@ -61,7 +61,7 @@ class DataSource(HashableBaseModel, ModelWithMetadataParsing):
     measures: Sequence[Measure] = []
     dimensions: Sequence[Dimension] = []
 
-    mutability: Mutability
+    mutability: Mutability = Mutability(type=MutabilityType.FULL_MUTATION, type_params=None)
```

Comment:
We currently define all of our defaults in our model objects, but our dev tooling efforts around IDE integration are most likely going to be based on the jsonschema. This seems a bit off to me, but for now let's leave it as-is.

Longer term we might consider getting to a place where we can generate the jsonschema from our Pydantic objects. We have type mismatches between the jsonschema and the pydantic object itself which makes that very difficult at the moment, but it's something to think about.